toolkit/offchain/src/register.rs

```diff
@@ -208,6 +295,30 @@ fn register_tx(
 	tx_builder.balance_update_and_build(ctx)
 }
 
+fn deregister_tx(
+	validator: &PlutusScript,
+	own_registration_utxos: &[OgmiosUtxo],
+	ctx: &TransactionContext,
+	validator_redeemer_ex_units: ExUnits,
+) -> Result<Transaction, JsError> {
+	let config = crate::csl::get_builder_config(ctx)?;
+	let mut tx_builder = TransactionBuilder::new(&config);
+
+	{
+		let mut inputs = TxInputsBuilder::new();
+		for own_registration_utxo in own_registration_utxos {
```

Comment:
In what situation would there be more than one registration utxo? A bug? I think re-registration burns the previous one, right? Anyway, I think burning all in deregister like this is the way to go.

Reply:
If it is a bug depends on the specification, which I do not know.
Mainchain follower component sorts registrations and regards only that latest one.

Reply:
For me clear is that if there were 2 matching utxos then we should spend both, because keeping registration(s) after deregister command is a bad behavior for me.

Reply:
Yes, it is